packages/salesforcedx-vscode-apex-replay-debugger/src/breakpoints/checkpointService.ts

```diff
@@ -644,24 +653,26 @@ export async function processBreakpointChangedForCheckpoints(
       const theNode = checkpointService.returnCheckpointNodeIfAlreadyExists(
         breakpointId
       );
-      // If the node exists then update it
-      if (theNode) {
-        theNode.updateCheckpoint(
-          bp.enabled,
-          uri,
-          filename,
-          checkpointOverlayAction
-        );
-      } else {
-        // else if the node didn't exist then create it
-        checkpointService.createCheckpointNode(
-          breakpointId,
-          bp.enabled,
-          uri,
-          filename,
-          checkpointOverlayAction
-        );
-      }
+      await lock.acquire(CHECKPOINTS_LOCK_STRING, async () => {
+        // If the node exists then update it
+        if (theNode) {
+          theNode.updateCheckpoint(
+            bp.enabled,
+            uri,
+            filename,
+            checkpointOverlayAction
+          );
+        } else {
+          // else if the node didn't exist then create it
+          checkpointService.createCheckpointNode(
+            breakpointId,
+            bp.enabled,
+            uri,
+            filename,
+            checkpointOverlayAction
+          );
+        }
+      });
     } else {
       // The breakpoint is no longer a SourceBreakpoint or is no longer a checkpoint. Call to delete it if it exists
       checkpointService.deleteCheckpointNodeIfExists(breakpointId);
```

Comment:
Could you explain why this doesn't need a lock?

Reply:
It does. Fixed in next push.

Reply:
It seems like everything in `processBreakpointChangedForCheckpoints` needs a lock. Can the whole method be locked instead of having multiple `lock.acquire`? Similar to what you did in `sfdxCreateCheckpoints`.

Reply:
Thoughts on ^^^?

Reply:
Not everything does. Remember that even non-checkpoint breakpoint changes will still go through processBreakpointChangedForCheckpoints. The guards here are if and only if it will require checkpoint changes. Breakpoints can still be added/removed and if they're not checkpoints then that won't stop in here.